internal/commands/build/build.go

```diff
@@ -138,6 +141,16 @@ func runBuild(dockerCli command.Cli, contextPath string, opt buildOptions) (refe
 		return nil, err
 	}
 
+	if opt.imageIDFile != "" {
+		id, err := store.FromBundle(bundle)
+		if err != nil {
+			return nil, err
```

Comment:
If there is an error here it means that the build passed but we couldn’t get the bundle, we should print a nice message like “couldn’t get the id” or something

Reply:
Not sure about this. This error would actually happen if we can't marshall the bundle to compute a sha256, which means either a major golang runtime failure, or some weirdness in the bundle data, and then the built image would also be invalid. Better imho not to pretend something minor went wrong here

Reply:
revisited as https://github.com/docker/app/pull/705/files#diff-03504424ddde121c663ff197df2ad5caR112-R114
ID is now returned when persisting in store, and on failure writting to iidfile an error is reported on stderr, while still confirming app image has been built